manheim_c7n_tools/dryrun_diff.py

```diff
@@ -100,6 +112,87 @@ def _find_changed_policies(self, git_dir=None, diff_against='master'):
             pnames.append(m.group(1))
         return pnames
 
+    def _make_diff_report(self, dryrun):
+        """
+        Return a HTML report breaking down the differences between the dryrun
+        (this branch) and the last run of each policy on master.
+
+        Requires a jinja template located at `./reporting-template/report.j2`
+        within the same directory where the dryrun-diff step has been run.
+
+        :param dryrun: dryrun policy resource information
+        :type dryrun: dict
+        :return: html report
+        :rtype: str
+        """
+        all_policies = list(
+            set(dryrun.keys()) | set(self._live_results.keys())
+        )
+        entries = {}
+        if not all_policies:
+            logger.info('no policies found - skipping diff report')
+            return ''
+        t_loader = FileSystemLoader(searchpath="./reporting-template/")
+        t_env = Environment(loader=t_loader)
+        t_file = "report.j2"
+        try:
+            tmpl = t_env.get_template(t_file)
+        except TemplateNotFound:
+            logger.info('unable to find a template - skipping diff report')
+            return ''
+
+        for policy in all_policies:
+            policy_splits = policy.split('-', 1)
+            if len(policy_splits) <= 1:
+                logger.warning(
+                    "skipping policy: \
+                        %s from report - it is not prefixed correctly",
+                    policy)
+                continue
+            resource_type = policy_splits[0]
+            load_available()
+            _id = get_resource_class(resource_type) \
+                .resource_type() \
+                .id
```

Comment:
This _has_ the potential to throw an exception if it cannot find the appropiate resource - perhaps it might be worth placing some guardrails around this...